### ISSUE TITLE
Sample produces RejectedExecutionException

### ISSUE THREAD

**Issue by eveliotc** [chrisbanes/Android-BitmapCache#6]
Hi,

Gave Sample a try on two Samsung Gingerbread phones got this exception after a while of scrolling grid up and down, have you considered using own thread pool instead of AsyncTasks?

Android 2.3.3 on GT-I9000 (Galaxy S)

``` java
02-21 03:23:46.320: ERROR/AndroidRuntime(7965): FATAL EXCEPTION: main
        java.util.concurrent.RejectedExecutionException
        at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:1876)
        at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:774)
        at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1295)
        at android.os.AsyncTask.execute(AsyncTask.java:394)
        at uk.co.senab.bitmapcache.samples.NetworkedCacheableImageView.loadImage(NetworkedCacheableImageView.java:158)
        at uk.co.senab.bitmapcache.samples.PugListAdapter.getView(PugListAdapter.java:61)
        at android.widget.AbsListView.obtainView(AbsListView.java:1933)
        at android.widget.GridView.makeAndAddView(GridView.java:1219)
        at android.widget.GridView.makeRow(GridView.java:265)
        at android.widget.GridView.fillUp(GridView.java:303)
        at android.widget.GridView.fillGap(GridView.java:196)
        at android.widget.AbsListView.trackMotionScroll(AbsListView.java:3992)
        at android.widget.AbsListView$FlingRunnable.run(AbsListView.java:3437)
        at android.os.Handler.handleCallback(Handler.java:587)
        at android.os.Handler.dispatchMessage(Handler.java:92)
        at android.os.Looper.loop(Looper.java:123)
        at android.app.ActivityThread.main(ActivityThread.java:4627)
        at java.lang.reflect.Method.invokeNative(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:521)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:860)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618)
        at dalvik.system.NativeStart.main(Native Method)
```

Android 2.3.6 on SPH-M820-BST (Prevail)

``` java
02-20 18:19:51.625: ERROR/AndroidRuntime(11323): FATAL EXCEPTION: main
        java.util.concurrent.RejectedExecutionException: pool=128/128, queue=10/10
        at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:1961)
        at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:794)
        at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1315)
        at android.os.AsyncTask.execute(AsyncTask.java:394)
        at uk.co.senab.bitmapcache.samples.NetworkedCacheableImageView.loadImage(NetworkedCacheableImageView.java:158)
        at uk.co.senab.bitmapcache.samples.PugListAdapter.getView(PugListAdapter.java:61)
        at android.widget.AbsListView.obtainView(AbsListView.java:1533)
        at android.widget.GridView.makeAndAddView(GridView.java:1269)
        at android.widget.GridView.makeRow(GridView.java:315)
        at android.widget.GridView.fillUp(GridView.java:353)
        at android.widget.GridView.fillGap(GridView.java:246)
        at android.widget.AbsListView.trackMotionScroll(AbsListView.java:4142)
        at android.widget.AbsListView$TwFlingRunnable.run(AbsListView.java:3370)
        at android.os.Handler.handleCallback(Handler.java:587)
        at android.os.Handler.dispatchMessage(Handler.java:92)
        at android.os.Looper.loop(Looper.java:123)
        at android.app.ActivityThread.main(ActivityThread.java:3687)
        at java.lang.reflect.Method.invokeNative(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:507)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:842)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:600)
        at dalvik.system.NativeStart.main(Native Method)
```

**Comment by curioustechizen** [comment]
I see the same problem with a Samsung Galaxy Tab (the original 7-inch one) running Android 2.3.3. Here's the stack trace:

```
java.util.concurrent.RejectedExecutionException: pool=128/128, queue=10/10
    at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:1961)
    at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:794)
    at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1315)
    at android.os.AsyncTask.execute(AsyncTask.java:394)
    at uk.co.senab.bitmapcache.samples.NetworkedCacheableImageView.loadImage(NetworkedCacheableImageView.java:158)
    at uk.co.senab.bitmapcache.samples.PugListAdapter.getView(PugListAdapter.java:61)
    at android.widget.AbsListView.obtainView(AbsListView.java:1533)
    at android.widget.GridView.makeAndAddView(GridView.java:1269)
    at android.widget.GridView.makeRow(GridView.java:315)
    at android.widget.GridView.fillDown(GridView.java:268)
    at android.widget.GridView.fillGap(GridView.java:235)
    at android.widget.AbsListView.trackMotionScroll(AbsListView.java:4142)
    at android.widget.AbsListView$TwFlingRunnable.run(AbsListView.java:3370)
    at android.os.Handler.handleCallback(Handler.java:587)
    at android.os.Handler.dispatchMessage(Handler.java:92)
    at android.os.Looper.loop(Looper.java:123)
    at android.app.ActivityThread.main(ActivityThread.java:3687)
    at java.lang.reflect.Method.invokeNative(Native Method)
    at java.lang.reflect.Method.invoke(Method.java:507)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:842)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:600)
    at dalvik.system.NativeStart.main(Native Method)
```

It appears to happen when we try to fetch too many images from the network at once.

**Comment by chrisbanes** [comment]
'Fixed' in ffef32129e0bfe883de19725595ae39157d7cf48.
